### PR TITLE
Julia 0.7

### PR DESCRIPTION
**Pull request by davidanthoff**
This is not enough to get things passing on 1.0, but it is a start. I'll have some more comments inline.

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,2 +1,2 @@
-julia 0.7-
-BinaryProvider 0.3.0
+julia 0.7
+BinaryProvider 0.4.0
```
Comment (davidanthoff) [inline REQUIRE]: This will have to be 0.4.1, once https://github.com/JuliaPackaging/BinaryProvider.jl/pull/97 is merged.

File: test/runtests.jl [hunk 2]
```diff
@@ -21,7 +21,7 @@ for (s, enc) in (("noël", "ISO-8859-1"),
 end
 
 # Test that attempt to close stream in the middle of incomplete sequence throws
-let s = "a string チャネルパートナーの選択", a = Vector{UInt8}(s)
+let s = "a string チャネルパートナーの選択", a = unsafe_wrap(Vector{UInt8}, s)
```
Comment (davidanthoff) [inline test/runtests.jl]: I think that is the right fix, but now tests fail...

**Comment by davidanthoff** [comment]
I need a working version of this package to get ReadStat.jl to work on 1.0. ReadStat.jl has a dep on this here, so that it will load the same IConv library as this package here. A bit of a hack, but I think the right approach until the BinaryProvider situation gets better.

File: test/runtests.jl (hunk 2 above)
Comment (nalimilan) [reply test/runtests.jl]: I don't think anything needs to be changed, IIRC this will now make a copy but it's OK.

**Comment by tk3369** [comment]
Same situation with SASLib.jl.

**Comment by davidanthoff** [comment]
@nalimilan could you restart the one failed build? I think it just timed out. Given that everything else passed, this PR is most likely actually ok.

File: REQUIRE (hunk 1 above)
Comment (nalimilan) [reply REQUIRE]: Need to change this?

File: appveyor.yml [hunk 3]
```diff
@@ -1,14 +1,22 @@
 environment:
   matrix:
-  - JULIA_URL: "https://julialang-s3.julialang.org/bin/winnt/x86/0.7/julia-0.7-latest-win32.exe"
-  - JULIA_URL: "https://julialang-s3.julialang.org/bin/winnt/x64/0.7/julia-0.7-latest-win64.exe"
-  - JULIA_URL: "https://julialangnightlies-s3.julialang.org/bin/winnt/x86/julia-latest-win32.exe"
-  - JULIA_URL: "https://julialangnightlies-s3.julialang.org/bin/winnt/x64/julia-latest-win64.exe"
+  - julia_version: 0.7
+  - julia_version: 1
```
Comment (nalimilan) [inline appveyor.yml]: I'd prefer 1.0 (there's a long thread about this somewhere).

File: appveyor.yml [hunk 4]
```diff
@@ -1,14 +1,22 @@
...
+  - julia_version: nightly
+
+platform:
+  - x86 # 32-bit
+  - x64 # 64-bit
+
+matrix:
+  allow_failures:
+  - julia_version: latest
```
Comment (nalimilan) [inline appveyor.yml]: Should this be `nightly`?

**Comment by davidanthoff** [comment]
@nalimilan Alright, I think ready to merge and then tag a release, right?

**Comment by nalimilan** [comment]
Yes, thanks!

**Comment by nalimilan** [comment]
https://github.com/JuliaLang/METADATA.jl/pull/17351/files

**Pull request merged**